AppAuth.podspec

```diff
@@ -31,7 +31,10 @@ It follows the OAuth 2.0 for Native Apps best current practice
   #       classes of AppAuth with tokens on watchOS and tvOS, but currently the
   #       library won't help you obtain authorization grants on those platforms.
 
-  s.platforms    = { :ios => "7.0", :osx => "10.9", :watchos => "2.0", :tvos => "9.0" }
+  s.ios.deployment_target = "7.0"
+  s.osx.deployment_target = "10.9"
+  s.watchos.deployment_target = "2.0"
+  s.tvos.deployment_target = "9.0"
```

Comment:
Are you sure this is equivalent? Can you point me to a doc that validates this?

Reply:
And if it is equivalent, is the line below still needed?
`externalUserAgent.osx.deployment_target = '10.9'`

Reply:
> Are you sure this is equivalent? Can you point me to a doc that validates this?

https://guides.cocoapods.org/syntax/podspec.html#platform : "When supporting multiple platforms you should use deployment_target below instead."

Reply:
> And if it is equivalent, is the line below still needed?
> `externalUserAgent.osx.deployment_target = '10.9'`

Here https://guides.cocoapods.org/syntax/podspec.html#group_platform it states that "If not defined in a subspec the attributes of this group inherit the value of the parent."

The deployment_targets specified in ExternalUserAgent and EnterpriseUserAgent state that for those subspecs, only osx and ios are supported (From my understanding)

Reply:
Thank you for confirming that, looks good.

I think we can delete the duplicate definitions in the subspecs in that case, there doesn't seem to be any added value by duplicating it.

Reply:
Actually on further reflection, are we sure the EnterpriseUserAgent subspec is not inheriting the tvOS and watchOS deployment targets? The "If not defined in a subspec the attributes of this group inherit the value of the parent." comment refers to the docs for `platform`, not the `deployment_target` so I'm not sure just from reading the docs how it behaves, can you verify with experimentation?

Reply:
"platform" and "deployment_target" are both subheadings under "Platform" in the docs, so I thought they are equivalents.

I made some test projects. Indeed, if I try to import AppAuth/ExternalUserAgent explicitly in a watchOS or tvOS app podfile, I get an error of this type: "[!] The platform of the target `watchsample` (watchOS 6.2) is not compatible with `AppAuth/ExternalUserAgent (1.4.0)`, which does not support `watchOS`."

However, the behavior is strange when I try to simply import 'AppAuth' in the podfile in watchOS/tvOS.

If `default_subspec = ExternalUserAgent`, then nothing is imported - no Core and no ExternalUserAgent libraries. the imported Pod is empty besides the podspec, readme, license, and xcconfigs.

if `default_subspec = Core, ExternalUserAgent` then when I import 'AppAuth', then the watch/tvOS app imports the Core libraries and AppAuthCore.h, but no ExternalUserAgent files.

Is this okay behavior? I updated the PR to have the two imports.

Reply:
Thanks for researching this!

The default subspec behavior sounds desirable to me.